### ISSUE TITLE
When Attackers are fleeing, the Defense is not won

### ISSUE THREAD

**Issue by Pingger** [Pingger/Empire-Mod#7]
Instead you have to hunt down all fleeing attackers. The Attackers are stuck at the map edges.

**Comment by FalconElaris** [comment]
Since you're not setting ```settlement.isUnderAttack = false``` until the ```endBattle``` method, and you only call that if defenderForce is null or if it's an AutoBattle (Reference ```startDefense``` method), could it be that the defense is still considered ongoing? Whether or not the attackers are fleeing is never checked (that I saw) to determine if the battle is won.

**Comment by Pingger** [comment]
might be. But I have yet to understand most of the military coding as it is a big mess...

**Comment by FalconElaris** [comment]
I tested this, and there's an infinite loop when the enemy raiders attempt to flee. Here's the error code that repeats when they reach the edge of the map: (Doplhin is the name of the pawn that was fleeing)

``Dolphin started 10 jobs in 10 ticks. List: (Wait_MaintainPosture (Job_24635)) , (Goto (Job_24637) A=(11, 0, 0)) (Wait_MaintainPosture (Job_24638)) , (Goto (Job_24639) A=(11, 0, 0)) (Wait_MaintainPosture (Job_24640)) , (Goto (Job_24641) A=(11, 0, 0)) (Wait_MaintainPosture (Job_24642)) , (Goto (Job_24643) A=(11, 0, 0)) (Wait_MaintainPosture (Job_24644)) , (Goto (Job_24645) A=(11, 0, 0)) (Wait_MaintainPosture (Job_24646))  curDriver=JobDriver_WaitMaintainPosture (toilIndex=0) curJob=(Wait_MaintainPosture (Job_24646))
UnityEngine.StackTraceUtility:ExtractStackTrace ()
Verse.Log:Error (string)
Verse.AI.JobUtility:TryStartErrorRecoverJob (Verse.Pawn,string,System.Exception,Verse.AI.JobDriver)
Verse.AI.Pawn_JobTracker:FinalizeTick ()
(wrapper dynamic-method) Verse.AI.Pawn_JobTracker:Verse.AI.Pawn_JobTracker.JobTrackerTick_Patch0 (Verse.AI.Pawn_JobTracker)
(wrapper dynamic-method) Verse.Pawn:Verse.Pawn.Tick_Patch1 (Verse.Pawn)
Verse.TickList:Tick ()
(wrapper dynamic-method) Verse.TickManager:Verse.TickManager.DoSingleTick_Patch1 (Verse.TickManager)
Verse.TickManager:TickManagerUpdate ()
(wrapper dynamic-method) Verse.Game:Verse.Game.UpdatePlay_Patch1 (Verse.Game)
Verse.Root_Play:Update ()```

**Comment by Pingger** [comment]
Well this will be tricky to find ... Since there isn't any Empire code referenced ...

**Comment by FalconElaris** [comment]
Yep. It's not ideal, but it might give insight. I don't have time right now, but I'll try to track down some more hints in the methods that are used to declare battles won in the base game. It might prove useful to check those.
